src/Waives.Http/WaivesClient.cs

```diff
@@ -76,27 +107,49 @@ public async Task<Classifier> GetClassifier(string name)
 
         public async Task<IEnumerable<Document>> GetAllDocuments()
         {
-            var response = await HttpClient.GetAsync("/documents").ConfigureAwait(false);
+            var request = new HttpRequestMessage(HttpMethod.Get, new Uri("/documents", UriKind.Relative));
+            var response = await SendRequest(request).ConfigureAwait(false);
             await EnsureSuccessStatus(response).ConfigureAwait(false);
 
             var responseContent = await response.Content.ReadAsAsync<DocumentCollection>().ConfigureAwait(false);
-            return responseContent.Documents.Select(d => new Document(this, d.Links));
+            return responseContent.Documents.Select(d => new Document(this, d.Links, d.Id));
         }
 
         public async Task Login(string clientId, string clientSecret)
         {
-            var response = await HttpClient.PostAsync("/oauth/token", new FormUrlEncodedContent(new Dictionary<string, string>
-            {
-                { "client_id", clientId },
-                { "client_secret", clientSecret },
-            })).ConfigureAwait(false);
+            Logger.Log(LogLevel.Info, $"Logging in to Waives at {HttpClient.BaseAddress}");
 
+            var request = new HttpRequestMessage(HttpMethod.Post, new Uri("/oauth/token", UriKind.Relative))
+            {
+                Content = new FormUrlEncodedContent(new Dictionary<string, string>
+                {
+                    {"client_id", clientId},
+                    {"client_secret", clientSecret}
+                })
+            };
+
+            var response = await SendRequest(request).ConfigureAwait(false);
             await EnsureSuccessStatus(response).ConfigureAwait(false);
 
             var responseContent = await response.Content.ReadAsAsync<AccessToken>().ConfigureAwait(false);
             var accessToken = responseContent.Token;
 
             HttpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {accessToken}");
+            Logger.Log(LogLevel.Info, "Logged in.");
+        }
+
+        internal async Task<HttpResponseMessage> SendRequest(HttpRequestMessage request)
+        {
+            var stopWatch = new Stopwatch();
+            Logger.Log(LogLevel.Trace, $"Sending {request.Method} request to {request.RequestUri}");
+
+            stopWatch.Start();
+            var response = await HttpClient.SendAsync(request).ConfigureAwait(false);
+            stopWatch.Stop();
+
+            Logger.Log(LogLevel.Trace, $"Received response from {request.Method} {request.RequestUri} ({response.StatusCode}) ({stopWatch.ElapsedMilliseconds} ms)");
```

Comment:
:muscle: I think it'd be more useful to have the numeric version of the status code printed here, eg by casting to an int.

Reply:
Agreed. There's a `ReasonPhrase` property on `HttpResponse` which is more appropriate to use for the text version of the response, too: `StatusCode` will stringify 404 to `NotFound`, for example, while `ReasonPhrase` will render `Not Found`.

Is there no elapsed time (or similar) property on `HttpResponse`? Would be nice not to have to do that timing directly in the library if we can avoid it.